modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -363,7 +339,6 @@ public void processOpts() {
             importMapping.put("LocalDate", "org.joda.time.LocalDate");
             importMapping.put("DateTime", "org.joda.time.DateTime");
         } else if (dateLibrary.startsWith("java8")) {
-            additionalProperties.put("java8", "true");
```

Comment:
Yes it's currently the same flag for java8 methods and java8 dates.
This has been changed on the 2.3.0 branch : java8 dates now use `jsr310` tag.
So this change should be done iin the 2.3.0 branch